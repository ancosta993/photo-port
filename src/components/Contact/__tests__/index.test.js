import React from 'react';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ContactForm from '..'

afterEach(cleanup);

describe('ContactForm is rendering', () => {
   it('renders', () => {
      render(<ContactForm />);
   });

   it('matches Snapshot', () => {
      const {asFragment} = render(<ContactForm />);
      expect(asFragment()).toMatchSnapshot();
   })

   it('prints title', () => {
      const {getByTestId} = render(<ContactForm />);
      expect(getByTestId('contact')).toHaveTextContent('Contact me');
      expect(getByTestId('subBtn')).toHaveTextContent('Submit');
   })

})