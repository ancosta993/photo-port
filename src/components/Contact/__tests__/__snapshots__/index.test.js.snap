// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm is rendering matches Snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="contact"
    >
      Contact me
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          deafultvalue=""
          name="name"
          type="text"
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email address:
        </label>
        <input
          deafultvalue=""
          name="email"
          type="email"
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          deafultvalue=""
          name="message"
          row="5"
        />
      </div>
      <button
        data-testid="subBtn"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
